Source/levels/drlg_l1.cpp

```diff
@@ -1259,14 +1259,20 @@ void PlaceMiniSetRandom(const Miniset &miniset, int rndper)
 Point SelectChamber()
 {
 	int chamber;
-	if (!HasChamber1)
+	if (HasChamber1 && HasChamber2 && HasChamber3) {
+		chamber = PickRandomlyAmong({ 1, 2, 3 });

```

Comment:
This is not quite the same as `GenerateRnd(3) + 1`. It's possible for the former code to result in `-2 + 1`/`-1` which effectively acts the same as picking chamber 2. This code picks chamber 1 instead since PickRandomlyAmong corrects the original bias for the buggy case where GenerateRnd returns a negative value.

Reply:
```suggestion
		chamber = GenerateRnd(3) + 1;
```